### DISCUSSION TITLE
Usage with Rust?

### DISCUSSION THREAD

**Discussion by MarcCoquand** [piku/piku#329] [Q&A] [answered]
Hey Piku people!

I'm trying to figure out how to best use this with a custom runtime. As I understand it, it seems that you can set up a Procfile that runs the install scripts. So in theory, I could simply ssh into the machine, install the rust dependencies and then it should work?  Or is there another way you're supposed to do it?

**Comment by chr15m** [comment, marked as answer]
That sounds about right. As long as the `piku` user has access to those deps and binaries it should just work. You can use the `release` rule in the Procfile to run your own custom stuff. Note that if you have any files belonging to a different runtime (such as package.json or requirements.txt) then those will get picked up instead.

I've been working on a PR to enable deploying custom runtimes in a way that bypasses any other detected runtimes but it's not ready for review yet.

**Comment by MarcCoquand** [comment]
Great! I'll try to give this a spin then. Thank you so much! :)

**Comment by zexa** [comment]
I don't want to compile the binaries on my poor raspberry pi. The poor thing ends up overheating. Instead, I'd like to compile the application using a custom target on my machine, and then move it to my rpi. Does piku support such a use case?

**Reply by rcarmo** [reply]
As long as you're fine with committing the binaries to the git repo you push to piku, yes--it does not care about what you're doing, it merely compiles things _if_ it finds the instructions to do so.
